shell/platform/tizen/external_texture_gl.h

```diff
@@ -46,7 +46,7 @@ class ExternalTextureGL {
 
  private:
   std::unique_ptr<ExternalTextureGLState> state_;
-  std::mutex mutex_;
+  std::recursive_mutex mutex_;
```

Comment:
Why do we need recursive_mutex?

Reply:
Any other methods using ```mutex_``` are not affected? It seems that the behaviors may be different from what was originally intended.

Reply:
A recursive mutex is  just like mutex, **but allows the same thread to acquire multiple levels of ownership over the mutex object.** Except for this, everything else is the same.
because method PopulateTextureWithIdentifier will call DestructionTbmSurface when tbm_surface not valid.
if we want to call each other member function, we need use recursive mutex, mutex not support it.

of course we could not use recursive mutex, need add a new function.

Reply:
> A recursive mutex is just like mutex, **but allows the same thread to acquire multiple levels of ownership over the mutex object.** Except for this, everything else is the same.

That's right, I'm exactly concerned about the exception. 
Like your last words, I think that it would be better to add new function

Reply:
Will acquiring the mutex recursively on the same thread cause any problem?

Reply:
no problem now. I just concerned about someone's mistake in the future. I just want to avoid complicated situations when dealing with situations that can cause deadlock as possible. please ignore my opinion if you don't want

Reply:
Yeah, avoiding reentrant locks is a good practice. If `DestructionTbmSurface` is always called in guarded contexts, can we avoid acquring `mutex_` inside `DestructionTbmSurface`?

Reply:
we can add a new method without mutex.